src/app/index.ts

```diff
@@ -96,18 +96,38 @@ async function init(config?: IAppConfig) {
                   .then((res) => ({ stats: res }))
             : import('virtual:stats'));
 
-        console.log(stats);
-
         const updateSplashData = (data: TAppImportMap) => {
-            /** @todo implement with stats data */
             function updateSplashWithStats() {
-                const total =
-                    1 + Object.keys(data.assets).length + Object.keys(data.components).length;
-                const items =
-                    (data.lang !== undefined ? 1 : 0) +
-                    Object.values(data.assets).filter((flag) => flag).length +
-                    Object.values(data.components).filter((flag) => flag).length;
-                updateSplash((items / total) * 100);
+                function getAssetsSize(all = true): number {
+                    return Object.entries(data.assets)
+                        .filter(([, val]) => all || val)
+                        .map(
+                            ([key]) =>
+                                stats.assets[assetManifest[key].path.split('assets/').slice(-1)[0]],
+                        )
+                        .reduce((a, b) => a + b, 0);
+                }
+
+                function getComponentsSize(all = true): number {
+                    return Object.entries(data.components)
+                        .filter(([, val]) => all || val)
+                        .map(([key]) => stats.modules[key])
+                        .reduce((a, b) => a + b, 0);
+                }
+
+                const allAssetsSize = getAssetsSize();
+                const loadedAssetsSize = getAssetsSize(false);
+
+                const allComponentsSize = getComponentsSize();
+                const loadedComponentsSize = getComponentsSize(false);
+
+                const allLangSize = stats.i18n.en as number;
```

Comment:
Wasn't sure how to detect the total size of the lang data which needed to be loaded and the loaded size.
`data.lang` is only giving the lang mode or undefined, so in this case, we can't preassume which lang data needed to be loaded, that's why I made the default `en` lang size as the `allLangSize`. But the problem is, now if the default lang has been selected as `es` and we need to load the `es` lang data, then the `allLangSize` and `loadedLangSize` data will mismatch. We probably need a better solution to handle this edge case @meganindya

Reply:
I made some changes to get around this.